concepts/references/about.md

```diff
@@ -1,3 +1,167 @@
 # About
 
-TODO
+To understand references it is necessary to understand ownership. Every value has a single owner, which is in a scope. When the scope is exited,
+the value is dropped. For example:
+
+```rust
+fn log(msg: String) { // msg takes ownership of the value passed to it
+    let formatted_datetime = String::new();
+    // code  to format current datetime snipped
+    println!("{} at {}", msg, formatted_datetime);
+} // msg is dropped here
+
+pub fn main() {
+    let my_string = "Something happened".to_string();
+    log(my_string); // passing my_string transfers ownership of its value to msg
+    //println!("{:?}", my_string); // uncommenting this line will error because my_string's value has been dropped while owned by msg
+}
+```
+
+A [reference][borrow] represents a _borrow_ of an owned value. Instead of transferring ownership of `my_string`, we can have `log` borrow it and
+then give it back to `my_string` in `main` at the end of `log`.
+
+To do this, we define `msg` as a reference by prefixing an ampersand. The type `&String` is read as "reference to a `String`".
+
+As `log`'s argument is now a reference, we need to pass a reference. To get a reference to `my_string`, prefix it with an ampersand, which is
+the [borrow operator](https://doc.rust-lang.org/reference/expressions/operator-expr.html#borrow-operators). This lends `my_string` to `log`
+for the duration of that function.
+
+```rust
+fn log(msg: &String) { //msg is defined as a reference with an ampersand
+    let formatted_datetime: String;
+    // code  to format current datetime snipped
+    println!("{} at {}", msg, formatted_datetime);
+}
+
+pub fn main() {
+    let my_string = "Something happened".to_string();
+    log(&my_string); // my_string is passed as a reference with an ampersand
+    println!("{:?}", my_string);
+}
+```
+
+## References are immutable by default
+
+A value at any given time can have multiple references to it as long as they are all immutable, as shown below
+
+```rust
+fn check_shapes(constant: &[u8], linear: &[u8], superlinear: &[u8]) -> (bool, bool, bool) {
+    (
+        is_constant(constant),       // an immutable reference
+        is_linear(linear),           // another immutable reference, which is okay
+        is_superlinear(superlinear), // okay, all three references are immutable
+    )
+}
```

Comment:
This isn't actually where the overlap occurs. As far as `check_shapes` is concerned, its three parameters are all independent.

The overlap is in the invocation, where overlapping sub-slices of `fibonacci` are passed to `check_shapes`. The whole point of `check_shapes` is to demonstrate that the overlapping references can all live simultaneously, it's not just rustc getting clever about dropping the reference as soon as a particular check is complete.

Reply:
Yes, you're absolutely right. My stamina is flagging and I was careless. I'll try to fix it before I knock off for the day.